polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -70,44 +70,43 @@ protected LocalPolarisMetaStoreManagerFactory(
 
   protected abstract PolarisMetaStoreSession createMetaStoreSession(
       @Nonnull StoreType store,
-      @Nonnull RealmId realmId,
+      @Nonnull Realm realm,
       @Nullable PolarisCredentialsBootstrap credentialsBootstrap,
       @Nonnull PolarisDiagnostics diagnostics);
 
   protected PrincipalSecretsGenerator secretsGenerator(
-      RealmId realmId, @Nullable PolarisCredentialsBootstrap credentialsBootstrap) {
+      Realm realm, @Nullable PolarisCredentialsBootstrap credentialsBootstrap) {
     if (credentialsBootstrap != null) {
-      return PrincipalSecretsGenerator.bootstrap(realmId.id(), credentialsBootstrap);
+      return PrincipalSecretsGenerator.bootstrap(realm.name(), credentialsBootstrap);
     } else {
       return PrincipalSecretsGenerator.RANDOM_SECRETS;
     }
   }
 
-  private void initializeForRealm(
-      RealmId realmId, PolarisCredentialsBootstrap credentialsBootstrap) {
+  private void initializeForRealm(Realm realm, PolarisCredentialsBootstrap credentialsBootstrap) {
     final StoreType backingStore = createBackingStore(diagnostics);
     sessionSupplierMap.put(
-        realmId.id(),
-        () -> createMetaStoreSession(backingStore, realmId, credentialsBootstrap, diagnostics));
+        realm.name(),
+        () -> createMetaStoreSession(backingStore, realm, credentialsBootstrap, diagnostics));
 
     PolarisMetaStoreManager metaStoreManager =
-        new PolarisMetaStoreManagerImpl(realmId, diagnostics, configurationStore, clock);
-    metaStoreManagerMap.put(realmId.id(), metaStoreManager);
+        new PolarisMetaStoreManagerImpl(realm, diagnostics, configurationStore, clock);
+    metaStoreManagerMap.put(realm.name(), metaStoreManager);
```

Comment:
Can this be keyed on `Realm` instead of `String`?

Reply:
I was think of the refactor of this part, but it seems a relative big refactor as it changes the key type of the map. Can we do that as a followup PR?